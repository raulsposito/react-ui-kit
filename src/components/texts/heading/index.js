import React from 'react'
import PropTypes from 'prop-types'

import MyPropTyes from '@global/propTypes'

import { H1, H2, H3, H4, H5, H6 } from './styles'

const Heading = ({ weight, size, text, type, align, color, children, marginBottom, disabled }) => {
  const Tag = (() => {
    switch (type) {
      case 'h2': return H2
      case 'h3': return H3
      case 'h4': return H4
      case 'h5': return H5
      case 'h6': return H6
      default: return H1
    }
  })()

  return (
    <Tag
      weight={weight}
      color={color}
      align={align}
      size={size}
      disabled={disabled}
      marginBottom={marginBottom}
    >
      {text || children}
    </Tag>
  )
}

Heading.propTypes = {
  text: PropTypes.string,
  type: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),
  align: PropTypes.oneOf(['left', 'center', 'right']),
  color: MyPropTyes.color,
  weight: MyPropTyes.weight,
  size: MyPropTyes.size,
  children: PropTypes.node,
  marginBottom: MyPropTyes.space,
  disabled: PropTypes.bool
}

export default Heading
