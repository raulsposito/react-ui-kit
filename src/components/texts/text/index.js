import React from 'react'
import PropTypes from 'prop-types'

import MyPropTyes from '@global/propTypes'

import { StyledText } from './styles'

const Text = ({ text, align, weight, color, size, children, marginBottom, disabled }) => (
  <StyledText weight={weight} size={size} color={color} align={align} marginBottom={marginBottom} disabled={disabled}>
    {text || children}
  </StyledText>
)

Text.propTypes = {
  text: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  align: PropTypes.oneOf(['left', 'center', 'right']),
  weight: MyPropTyes.weight,
  color: MyPropTyes.color,
  size: MyPropTyes.size,
  children: PropTypes.node,
  marginBottom: MyPropTyes.space,
  disabled: PropTypes.bool
}

export default Text
