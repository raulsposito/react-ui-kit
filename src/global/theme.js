// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
