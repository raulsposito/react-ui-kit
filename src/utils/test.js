export const getComponents = styled => {
  const componentClass = styled.styledComponentId
  return document.getElementsByClassName(componentClass)
}

export const getComponent = (styled, index = 0) => {
  const components = getComponents(styled)
  return components[index]
}
