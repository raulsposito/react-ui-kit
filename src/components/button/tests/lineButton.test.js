import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { colors, spacing } from '@global/theme'
import { getComponent } from '@utils/test'

import { Text } from '@components/texts'
import Icon from '@components/icon'
import { LineButton } from '..'

const mockData = {
  text: 'help',
  icon: 'face',
  onClick: jest.fn()
}

describe('<LineButton />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<LineButton onClick={mockData.onClick} text={mockData.text} />)
    expect(asFragment()).toMatchSnapshot()
  })

  describe('Text and Icon', () => {
    it('has only text', () => {
      render(<LineButton onClick={mockData.onClick} text={mockData.text} />)
      const t = screen.queryByText(mockData.text)
      const i = getComponent(Icon)

      expect(t).toBeInTheDocument()
      expect(i).toBeUndefined()
    })

    it('has only icon', () => {
      render(<LineButton onClick={mockData.onClick} icon={mockData.icon} />)
      const t = getComponent(Text)
      const i = screen.queryByText(mockData.icon)

      expect(t).toBeUndefined()
      expect(i).toBeInTheDocument()
    })

    it('has both', () => {
      const { container } = render(
        <LineButton onClick={mockData.onClick} text={mockData.text} icon={mockData.icon} />
      )
      const t = screen.queryByText(mockData.text)
      const i = screen.queryByText(mockData.icon)

      expect(t).toBeInTheDocument()
      expect(i).toBeInTheDocument()

      const { childNodes } = container.firstChild
      expect(childNodes[0].textContent).toEqual(mockData.text)
      expect(childNodes[1].textContent).toEqual(mockData.icon)
    })

    it('has both in reverse order', () => {
      render(<LineButton onClick={mockData.onClick} text={mockData.text} icon={mockData.icon} reverse />)
      const b = screen.getByRole('button')
      expect(b).toHaveStyle({ 'flex-direction': 'row-reverse' })
    })
  })

  describe('Sizes', () => {
    it('has small size', () => {
      render(<LineButton onClick={mockData.onClick} size='small' text={mockData.text} />)

      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ height: `${spacing.threeAndAHalf}` })
      expect(button).toHaveStyle({ padding: `${spacing.threeQuarters} ${spacing.oneAndAHalf}` })
    })

    it('has medium size', () => {
      render(<LineButton onClick={mockData.onClick} size='medium' text={mockData.text} />)

      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ height: `${spacing.five}` })
      expect(button).toHaveStyle({ padding: `${spacing.one} ${spacing.two}` })
    })

    it('has large size', () => {
      render(<LineButton onClick={mockData.onClick} size='large' text={mockData.text} />)

      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ height: `${spacing.fiveAndAHalf}` })
      expect(button).toHaveStyle({ padding: `${spacing.oneAndAHalf} ${spacing.twoAndAHalf}` })
    })
  })

  describe('Block', () => {
    it('has width 100%', () => {
      render(<LineButton onClick={mockData.onClick} block text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ width: '100%' })
    })

    it('is not a block', () => {
      render(<LineButton onClick={mockData.onClick} text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ width: 'max-content' })
    })
  })

  describe('Disabled', () => {
    it('is disabled', () => {
      render(<LineButton onClick={mockData.onClick} disabled text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).toHaveAttribute('disabled')
      expect(button).toHaveStyle({
        cursor: 'inherit',
        'background-color': `${colors.background.disabled}`,
        'border-color': `${colors.primary.disabled}`
      })
    })

    it('is not disabled', () => {
      render(<LineButton onClick={mockData.onClick} text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).not.toHaveAttribute('disabled')
    })
  })

  describe('Fixed width', () => {
    describe('small', () => {
      it('does not has a fix width', () => {
        render(<LineButton onClick={mockData.onClick} size='small' />)
        const button = screen.getByRole('button')
        expect(button).toHaveStyle({ 'min-width': '' })
      })

      it('has fixed width', () => {
        render(<LineButton onClick={mockData.onClick} fixWidth size='small' />)
        const button = screen.getByRole('button')
        expect(button).toHaveStyle({ 'min-width': `${spacing.eight}` })
      })
    })

    describe('medium', () => {
      it('does not has a fix width', () => {
        render(<LineButton onClick={mockData.onClick} />)
        const button = screen.getByRole('button')
        expect(button).toHaveStyle({ 'min-width': '' })
      })

      it('has fixed width', () => {
        render(<LineButton onClick={mockData.onClick} fixWidth size='medium' />)
        const button = screen.getByRole('button')
        expect(button).toHaveStyle({ 'min-width': `${spacing.eleven}` })
      })
    })

    describe('large', () => {
      it('does not has a fix width', () => {
        render(<LineButton onClick={mockData.onClick} size='large' />)
        const button = screen.getByRole('button')
        expect(button).toHaveStyle({ 'min-width': '' })
      })

      it('has fixed width', () => {
        render(<LineButton onClick={mockData.onClick} fixWidth size='large' />)
        const button = screen.getByRole('button')
        expect(button).toHaveStyle({ 'min-width': `${spacing.fifteen}` })
      })
    })
  })

  describe('Colors', () => {
    it('has default color', () => {
      render(<LineButton onClick={mockData.onClick} text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ border: `2px solid ${colors.primary.regular}` })

      const t = screen.getByText(mockData.text)
      expect(t).toHaveStyle({ color: `${colors.primary.regular}` })
    })
    it('is primary', () => {
      render(<LineButton onClick={mockData.onClick} color='primary' text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ border: `2px solid ${colors.primary.regular}` })

      const t = screen.getByText(mockData.text)
      expect(t).toHaveStyle({ color: `${colors.primary.regular}` })
    })

    it('is variant', () => {
      render(<LineButton onClick={mockData.onClick} color='variant' text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ border: `2px solid ${colors.variant.regular}` })

      const t = screen.getByText(mockData.text)
      expect(t).toHaveStyle({ color: `${colors.variant.regular}` })
    })

    it('is secondary', () => {
      render(<LineButton onClick={mockData.onClick} color='secondary' text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ border: `2px solid ${colors.secondary.regular}` })

      const t = screen.getByText(mockData.text)
      expect(t).toHaveStyle({ color: `${colors.secondary.regular}` })
    })

    it('is success', () => {
      render(<LineButton onClick={mockData.onClick} color='success' text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ border: `2px solid ${colors.success.regular}` })

      const t = screen.getByText(mockData.text)
      expect(t).toHaveStyle({ color: `${colors.success.regular}` })
    })

    it('is error', () => {
      render(<LineButton onClick={mockData.onClick} color='error' text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ border: `2px solid ${colors.error.regular}` })

      const t = screen.getByText(mockData.text)
      expect(t).toHaveStyle({ color: `${colors.error.regular}` })
    })

    it('is system', () => {
      render(<LineButton onClick={mockData.onClick} color='system' text={mockData.text} />)
      const button = screen.getByRole('button')
      expect(button).toHaveStyle({ border: `2px solid ${colors.system.regular}` })

      const t = screen.getByText(mockData.text)
      expect(t).toHaveStyle({ color: `${colors.system.regular}` })
    })
  })

  describe('Mouse events', () => {
    it('calls function on click', () => {
      render(<LineButton onClick={mockData.onClick} />)
      const button = screen.getByRole('button')

      userEvent.click(button)
      expect(mockData.onClick).toHaveBeenCalledTimes(1)
    })
  })
})
