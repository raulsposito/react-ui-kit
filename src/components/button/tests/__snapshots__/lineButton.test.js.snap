// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineButton /> should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-fubCzh kBlhdU"
    type="line"
  >
    <p
      class="sc-bdfBQB iDqXfG"
    >
      help
    </p>
  </button>
</DocumentFragment>
`;
